dal/src/main/java/com/epam/esm/dal/CertificateDao.java

```diff
@@ -0,0 +1,21 @@
+package com.epam.esm.dal;
+
+import java.util.List;
+
+import com.epam.esm.entity.GiftCertificate;
+import com.epam.esm.transferobj.FilterParam;
+import com.epam.esm.transferobj.OrderParam;
+
+public interface CertificateDao {
+
+	GiftCertificate addCertificate(GiftCertificate certificate);
+
+	GiftCertificate updateCertificate(GiftCertificate certificate);
+
+	List<GiftCertificate> findCertificates(List<FilterParam> filterParams, List<OrderParam> orderParams);
+
+	GiftCertificate findCertificate(long id);
+
+	int[] deleteCertificate(long id);
```

Comment:
I think, it quite strange to return array of numbers, it's not obvious what they mean